docs/source/manual/file_formats/gsb_file_format.rst

```diff
@@ -0,0 +1,103 @@
+.. _file_formats_gsb_xml:
+
+GSB- General Switch Box (.xml)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``write_gsb_to_xml`` command writes out the genreal switch box information in 
+``.xml`` file format which allows designer to debug connectivity and connection 
+related issue. If ``--unique`` option is enabled in the command, 
+information related to each unique connection box and switch box is written out
+in file a ``cb?_?__?_gsb.xml`` and ``sb_?__?_gsb.xml`` respectively.
+Format of each file is detailed below.
+
+Switch Box GSB (.xml)
+---------------------
+.. code-block:: xml
+    
+    <rr_gsb x="LOC_X" y="LOC_Y" TYPE="SB" num_sides="SIDE">
+        <CHANY side="SIDE" index="INDX" node_id="nID" segment_id="sID" mux_size="mSIZE">
+            <driver_node type="TYPE" side="SIDE" index="INDX" node_id="nID" grid_side="gSIDE"/>
+            ..
+        </CHANY>
+        <INPUT CHANX="" CHANY="" OPIN="">
+            <driver_node type="CHANX" side="SIDE" index="INDX" node_id="nID" segment_id="sID"/>
+            <driver_node type="CHANY" side="SIDE" index="INDX" node_id="nID" segment_id="sID"/>
+            <driver_node type="OPIN" side="SIDE" index="INDX" node_id="nID" grid_side="gSIDE"/>
+        </INPUT>
+    </rr_gsb>
+
+
+.. option:: <rr_gsb x="LOC_X" y="LOC_Y" TYPE="SB" num_sides="SIDE">
+
+  - ``LOC_X`` X coordinate of the switch box in 2D FPGA grid
+  - ``LOC_Y`` Y coordinate of the switch box in 2D FPGA grid
+  - ``mSIZE`` Number of input signals multiplexes to generate the given output channel
+  - ``SIDE`` Not Sure
```

Comment:
Number of sides (TOP, RIGHT, BOTTOM, LEFT) available in a RSG. Most of time, it is 4.